web/src/main/webapp/heavyhittersview.ts

```diff
@@ -0,0 +1,168 @@
+        else
+            restPos = tdv.rows.length;
+
+        let tBody = table.createTBody();
+        if (tdv.rows != null) {
+            let k = 0;
+            let position = 0;
+            for (let i = 0; i < tdv.rows.length; i++) {
+                k++;
+                if (i == restPos) {
+                    this.showRest(k, position, restCount, tdv.rowCount, tBody);
+                    position += restCount;
+                    k++;
+                }
+                let trow = tBody.insertRow();
+                let cell = trow.insertCell(0);
+                cell.style.textAlign = "right";
+                cell.textContent = k.toString();
+                for (let j = 0; j < this.schema.length; j++) {
+                    let cell = trow.insertCell(j+1);
+                    cell.style.textAlign = "right";
+                    let value = tdv.rows[i].values[j];
+                    if (value == null) {
+                        cell.classList.add("missingData");
+                        cell.textContent = "missing";
+                    } else
+                        cell.textContent = TableView.convert(value, this.schema[j].kind);
+                }
+                let cell1 = trow.insertCell(this.schema.length + 1);
+                cell1.style.textAlign = "right";
+                cell1.textContent = String(tdv.rows[i].count);
+                let cell2 = trow.insertCell(this.schema.length + 2);
+                cell2.style.textAlign = "right";
+                cell2.textContent = significantDigits((tdv.rows[i].count/tdv.rowCount)*100);
+                let cell3 = trow.insertCell(this.schema.length + 3);
+                let dataRange = new DataRange(position, tdv.rows[i].count, tdv.rowCount);
+                cell3.appendChild(dataRange.getDOMRepresentation());
+                position += tdv.rows[i].count;
+            }
+            if ((restPos == tdv.rows.length) && (restCount > 0)) {
+                k = tdv.rows.length + 1;
+                this.showRest(k, position, restCount, tdv.rowCount, tBody);
+            }
+        }
+    }
+
+    private getCount(tdv:TableDataView): number{
+        if (tdv.rows == null)
+            return tdv.rowCount;
+        else {
+            let runCount = tdv.rowCount;
+            for (let i = 0; i < tdv.rows.length; i++)
+                runCount -= tdv.rows[i].count;
+            return runCount;
+        }
+    }
+
+    private getPos(tdv:TableDataView, restCount: number): number{
+        if (tdv.rows == null)
+            return 0;
+        else {
+            let i = 0;
+            while((i < tdv.rows.length) && (restCount <= tdv.rows[i].count))
+                i++;
+            return i;
+        }
+    }
+
+    private showRest(k: number, position: number, restCount: number, total: number, tBody: any): void {
+        let trow = tBody.insertRow();
+        let cell = trow.insertCell(0);
+        cell.style.textAlign = "right";
```

Comment:
Try to also use the "missingData" class (as above); this will color the contents red.